core-common/src/main/java/org/zalando/nakadi/repository/kafka/AvroDeserializerWithSequenceDecoder.java

```diff
@@ -27,20 +28,29 @@ public AvroDeserializerWithSequenceDecoder(final AvroSchema schemas) {
 
     public byte[] deserializeAvro(final EnvelopeHolder envelope) throws RuntimeException {
         try {
+            final byte metadataVersion = envelope.getMetadataVersion();
+
             final SequenceDecoder metadataDecoder = metadataSequenceDecoders.computeIfAbsent(
-                    String.valueOf(envelope.getMetadataVersion()),
+                    String.valueOf(metadataVersion),
                     (v) -> new SequenceDecoder(schemas.getEventTypeSchema(AvroSchema.METADATA_KEY, v)));
 
             final GenericRecord metadata = metadataDecoder.read(envelope.getMetadata());
 
-            metadata.put("occurred_at", new DateTime(
-                    (long) metadata.get("occurred_at"), DateTimeZone.UTC).toString());
-            metadata.put("received_at", new DateTime(
-                    (long) metadata.get("received_at"), DateTimeZone.UTC).toString());
+            metadata.put(GenericRecordMetadata.OCCURRED_AT, new DateTime(
+                    (long) metadata.get(GenericRecordMetadata.OCCURRED_AT), DateTimeZone.UTC).toString());
+
+            metadata.put(GenericRecordMetadata.RECEIVED_AT, new DateTime(
+                    (long) metadata.get(GenericRecordMetadata.RECEIVED_AT), DateTimeZone.UTC).toString());
+
+            final String eventType = metadata.get(GenericRecordMetadata.EVENT_TYPE).toString();
+
+            final String schemaVersionField =
+                    metadataVersion < 4 ? "schema_version": GenericRecordMetadata.SCHEMA_VERSION;
```

Comment:
Do we really support versions that use "schema_version"? Since we use the metadata schema for only internal events, we could even reset the version to 0 and have the current state.

Reply:
> we could even reset the version to 0 and have the current state

I thought that we can drop versions older than "3" (and corresponding tests), as I find it is getting very messy.